### PR TITLE
SONARXML-127 Move rule S3419 from Java to XML Analyzer

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/maven/GroupIdNamingConventionCheck.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonarsource.analyzer.commons.annotations.DeprecatedRuleKey;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.sonarsource.analyzer.commons.xml.checks.SimpleXPathBasedCheck;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+@Rule(key = GroupIdNamingConventionCheck.KEY)
+@DeprecatedRuleKey(repositoryKey = "java", ruleKey = GroupIdNamingConventionCheck.KEY)
```
Comment (dorian-burihabwa-sonarsource) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/maven/GroupIdNamingConventionCheck.java:33]: The annotation needs to be removed

File: sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlRulesDefinitionTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 class XmlRulesDefinitionTest {
 
   private static final Set<String> JAVA_DEPRECATED_KEYS = Stream
-    .of("S3281", "S3355", "S3373", "S3822")
+    .of("S3281", "S3355", "S3373", "S3419", "S3822")
```
Comment (dorian-burihabwa-sonarsource) [inline sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlRulesDefinitionTest.java:38]: The key should be removed from deprecated keys for now

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
A few changes are needed around the use of the deprecation keys

**Comment by quentin-jaquier-sonarsource** [comment]
Move of all rules will be done at the same time, see branch cd53bc09b810420de0e844eb78e6c1ebdc643fbc.

**Pull request closed without merging**